File: WYNKWeb/src/app/Models/ViewModels/RoomMasterViewM.ts
import { Room } from '../Room';


export class RoomMasterViewM {

  RoomMaster: Room = new Room();

  GetRoomDet: Array<GetRoomDet> = [];
  RoomDetails1: Array<RoomDetails1> = [];
  RoomDetails2: Array<RoomDetails2> = [];
  RoomNoArr: Array<RoomNoArr> = [];
  BedNoarr: Array<BedNoarr> = [];
  ToiletType: Array<ToiletType> = [];
}



export class RoomNoArr {
  RNo: number;
  RToiletType: string;
}

export class BedNoarr {
  BedNo: string;
}

export class GetRoomDet {

  No: number;
  ToiletType: string;
  BedNo: string;
  RoomType: string;
  RoomDescription: string;
  RoomCost:number;
  
}
export class ToiletType
{
  RoomNo: number;
  RestRoomType: string;

}
export class RoomDetails1 {
  No: number;
  NoofBeds: number;
  ToiletType: string;
  RoomDetID: number;
}

export class RoomDetails2 {
  No: number;
  BedNo: string;
  ToiletType: string;
  RoomDetID: number;
  IsActive: string;
  RoomType: string;
  RoomDescription: string;
  Roomcost: number;
}

File: WYNK.Data/Model/ViewModel/RoomMasterViewM .cs
using System;
using System.Collections.Generic;
using System.Text;

namespace WYNK.Data.Model.ViewModel
{
    public class RoomMasterViewM
    {
        public Room RoomMaster { get; set; }
        public RoomDetails RoomDetails { get; set; }

        public ICollection<getRoomDet> getRoomDet { get; set; }
        public ICollection<RoomDetails1> RoomDetails1 { get; set; }
        public ICollection<RoomDetails2> RoomDetails2 { get; set; }
        public ICollection<RoomTariff> RoomTariff { get; set; }

        public ICollection<ToiletType> ToiletType { get; set; }
        
    }
    public class RoomTariff
    {
        public string RoomType { get; set; }
        public string RoomDescription { get; set; }
        public decimal Roomcost { get; set; }
        
        public string PAddress { get; set; }
        public string PAddress2 { get; set; }
        public string PAddress3 { get; set; }
        public string Pphone { get; set; }
        public string Pweb { get; set; }
        public string PCompnayname { get; set; }
    }
    public class getRoomDet
    {
        public int No { get; set; }
        public object ToiletType { get; set; }
        public string BedNo { get; set; }

    }
    public class ToiletType
    {
        public int RoomNo { get; set; }
        public string RestRoomType { get; set; }

    }
    public class RoomDetails1
    {

        public int No { get; set; }
        public string ToiletType { get; set; }
        public string BedNo { get; set; }
        public string RoomType { get; set; }
        public string RoomDescription { get; set; }
        public decimal Roomcost { get; set; }
        public string IsActive { get; set; }
        public int RoomDetID { get; set; }

    }
    public class RoomDetails2
    {

        public int No { get; set; }
        public string ToiletType { get; set; }
        public string BedNo { get; set; }
        public string RoomType { get; set; }
        public string RoomDescription { get; set; }
        public decimal Roomcost { get; set; }
        public string IsActive { get; set; }
        public int RoomDetID { get; set; }

    }

}
